common/metrics/metric_defs.go

```diff
@@ -77,359 +75,29 @@ const (
-	// AdminClientRemoveRemoteClusterScope tracks RPC calls to admin service
-	AdminClientRemoveRemoteClusterScope = "AdminClientRemoveRemoteCluster"
-	// AdminClientGetReplicationMessagesScope tracks RPC calls to admin service
-	AdminClientGetReplicationMessagesScope = "AdminClientGetReplicationMessages"
-	// AdminClientGetNamespaceReplicationMessagesScope tracks RPC calls to admin service
-	AdminClientGetNamespaceReplicationMessagesScope = "AdminClientGetNamespaceReplicationMessages"
-	// AdminClientGetDLQReplicationMessagesScope tracks RPC calls to admin service
-	AdminClientGetDLQReplicationMessagesScope = "AdminClientGetDLQReplicationMessages"
-	// AdminClientReapplyEventsScope tracks RPC calls to admin service
-	AdminClientReapplyEventsScope = "AdminClientReapplyEvents"
-	// AdminClientGetDLQMessagesScope tracks RPC calls to admin service
-	AdminClientGetDLQMessagesScope = "AdminClientGetDLQMessages"
-	// AdminClientPurgeDLQMessagesScope tracks RPC calls to admin service
-	AdminClientPurgeDLQMessagesScope = "AdminClientPurgeDLQMessages"
-	// AdminClientMergeDLQMessagesScope tracks RPC calls to admin service
-	AdminClientMergeDLQMessagesScope = "AdminClientMergeDLQMessages"
-	// AdminClientRefreshWorkflowTasksScope tracks RPC calls to admin service
-	AdminClientRefreshWorkflowTasksScope = "AdminClientRefreshWorkflowTasks"
-	// AdminClientResendReplicationTasksScope tracks RPC calls to admin service
-	AdminClientResendReplicationTasksScope = "AdminClientResendReplicationTasks"
-	// AdminClientGetTaskQueueTasksScope tracks RPC calls to admin service
-	AdminClientGetTaskQueueTasksScope = "AdminClientGetTaskQueueTasks"
-	// AdminClientDeleteWorkflowExecutionScope tracks RPC calls to admin service
-	AdminClientDeleteWorkflowExecutionScope = "AdminClientDeleteWorkflowExecution"
 
-	// AdminDescribeHistoryHostScope is the metric scope for admin.AdminDescribeHistoryHost
-	AdminDescribeHistoryHostScope = "AdminDescribeHistoryHost"
-	// AdminAddSearchAttributesScope is the metric scope for admin.AdminAddSearchAttributes
-	AdminAddSearchAttributesScope = "AdminAddSearchAttributes"
-	// AdminRemoveSearchAttributesScope is the metric scope for admin.AdminRemoveSearchAttributes
-	AdminRemoveSearchAttributesScope = "AdminRemoveSearchAttributes"
-	// AdminGetSearchAttributesScope is the metric scope for admin.AdminGetSearchAttributes
-	AdminGetSearchAttributesScope = "AdminGetSearchAttributes"
-	// AdminRebuildMutableStateScope is the metric scope for admin.AdminRebuildMutableState
-	AdminRebuildMutableStateScope = "AdminRebuildMutableState"
-	// AdminDescribeMutableStateScope is the metric scope for admin.AdminDescribeMutableState
-	AdminDescribeMutableStateScope = "AdminDescribeMutableState"
 	// AdminGetWorkflowExecutionRawHistoryV2Scope is the metric scope for admin.GetWorkflowExecutionRawHistoryScope
 	AdminGetWorkflowExecutionRawHistoryV2Scope = "AdminGetWorkflowExecutionRawHistoryV2"
```

Comment:
It looks like this one is used in the actual handler to record metrics.ServiceLatency, but that should be handled by TelemetryInterceptor. Could you double-check that and remove the latency code from there?

I see there's one more use in that handler, which is for metrics.HistorySize. I guess we can keep this constant just for that

Reply:
Sorry, re-opening this PR after a long lapse. Can you elaborate on this? I think because the code was updated since this, it's a bit harder to understand.

Reply:
Yeah, I think I saw a PR go by from @yycptt that fixed this double-recording